cupy/linalg/norms.py

```diff
@@ -268,13 +272,17 @@ def slogdet(a):
 
     diag = cupy.diagonal(lu, axis1=-2, axis2=-1)
 
+    logdet = cupy.log(cupy.abs(diag)).sum(axis=-1)
+
     # ipiv is 1-origin
-    non_zero = (cupy.count_nonzero(ipiv != cupy.arange(1, n + 1), axis=-1) +
-                cupy.count_nonzero(diag < 0, axis=-1))
+    non_zero = cupy.count_nonzero(ipiv != cupy.arange(1, n + 1), axis=-1)
+    if dtype.kind == "f":
+        non_zero += cupy.count_nonzero(diag < 0, axis=-1)
 
     # Note: sign == -1 ** (non_zero % 2)
     sign = (non_zero % 2) * -2 + 1
-    logdet = cupy.log(abs(diag)).sum(axis=-1)
+    if dtype.kind == "c":
+        sign = sign * cupy.prod(diag / cupy.abs(diag), axis=-1)
```

Comment:
You can also do this in-place:
```suggestion
        sign *= cupy.prod(diag / cupy.abs(diag), axis=-1)
```

Reply:
Thank you for your suggestions! :)

Unfortunately, this operation cannot be done because the `cupy.prod(...)` has complex numbers and the original `sign` contains real numbers.

Reply:
I see, thanks!